packages/adapters/solana/src/client.ts

```diff
@@ -1,699 +1,457 @@
-          throw new Error('Chain namespace is not supported')
-        }
-
-        const connection = SolStoreUtil.state.connection
-
-        if (!connection || !this.provider) {
-          throw new Error('Connection is not set')
-        }
+    // Add Coinbase Wallet if available
+    if ('coinbaseSolana' in window) {
+      this.addConnector({
+        id: 'coinbaseWallet',
+        type: 'EXTERNAL',
+        // @ts-expect-error window.coinbaseSolana exists
+        provider: new CoinbaseWalletProvider({
+          provider: window.coinbaseSolana as SolanaCoinbaseWallet,
+          chains: this.caipNetworks as CaipNetwork[],
+          getActiveChain: () => appKit.getCaipNetwork(this.namespace) as CaipNetwork
+        }),
+        name: 'Coinbase Wallet',
+        chain: this.namespace as ChainNamespace,
+        chains: []
+      })
+    }
 
-        const transaction = await createSendTransaction({
-          provider: this.provider,
-          connection,
-          to: '11111111111111111111111111111111',
-          value: 1
+    // Watch for standard wallet adapters
+    watchStandard(
+      this.caipNetworks as CaipNetwork[],
+      () => appKit.getCaipNetwork(this.namespace),
+      (...providers: WalletStandardProvider[]) => {
+        providers.forEach(provider => {
+          this.addConnector({
+            id: provider.name,
+            type: 'ANNOUNCED',
+            provider: provider as unknown as Provider,
+            imageUrl: provider.icon,
+            name: provider.name,
+            chain: CommonConstantsUtil.CHAIN.SOLANA,
+            chains: []
+          })
         })
+      }
+    )
+  }
 
-        const fee = await transaction.getEstimatedFee(connection)
-
-        return BigInt(fee || 0)
-      },
-      // -- Transaction methods ---------------------------------------------------
-      /**
-       *
-       * These methods are supported only on `wagmi` and `ethers` since the Solana SDK does not support them in the same way.
-       * These function definition is to have a type parity between the clients. Currently not in use.
-       */
-      getEnsAvatar: async (value: string) => await Promise.resolve(value),
-
-      getEnsAddress: async (value: string) => await Promise.resolve(value),
-
-      writeContract: async () => await Promise.resolve('0x'),
-
-      getCapabilities: async () => await Promise.resolve('0x'),
+  // -- Transaction methods ---------------------------------------------------
+  /**
+   *
+   * These methods are supported only on `wagmi` and `ethers` since the Solana SDK does not support them in the same way.
+   * These function definition is to have a type parity between the clients. Currently not in use.
+   */
+  // eslint-disable-next-line @typescript-eslint/require-await
+  public async getEnsAddress(
+    params: AdapterBlueprint.GetEnsAddressParams
+  ): Promise<AdapterBlueprint.GetEnsAddressResult> {
+    return { address: params.name }
+  }
```

Comment:
we should probably abstract a bit further (different PR) so that these things are kind of an extension to the blueprint rather than on the blueprint itself